### PR TITLE
User selection permissions

### LINKED ISSUE #217: Protect user searches
* certain users should only be able to search for other users on rental#new. 
* admin's should be able to search for everyone

### PR DESCRIPTION
**Pull request by tomecho**
fixes #217

### CODE REVIEW

**Comment by tomecho** [comment]
tests pass but im not sure if this works ~Dont Merge~

File: spec/controllers/rentals_controller_spec.rb [hunk 1]
```diff
@@ -75,6 +75,33 @@
...
+        User.destroy_all
+        @dept_one = create :department
+        @dept_one_users = create_list :user, 10, department: @dept_one
+        @other_users = create_list :user, 10 # not in @dept_one
+      end
+
+      it 'assigns all users if that user has the permission' do
+        u = @other_users.first
+        g = create(:group)
+        g.permissions << create(:permission, controller: 'rentals', action: 'assign_anyone')
```
Comment (owlee) [inline spec/controllers/rentals_controller_spec.rb]: Side note: We should create a function called create_with_permission/s inside group.

File: spec/controllers/rentals_controller_spec.rb [hunk 2]
```diff
@@ -75,6 +75,33 @@
...
+        g.save
+        u.groups << g
+        u.save
+        current_user(u) # set current_user to u in teh controller
+        get :new
+        expect(assigns[:users].collect {|user| user[:id] }).to match_array (@dept_one_users + @other_users).collect(&:id)
+      end
+
+      it 'it only assigns users in the same dept if they do not have the special permission' do
```
Comment (owlee) [inline spec/controllers/rentals_controller_spec.rb]: Typo, it 'it'

File: spec/controllers/rentals_controller_spec.rb [hunk 3]
```diff
@@ -75,6 +75,33 @@
...
+      end
+
+      it 'assigns all users if that user has the permission' do
+        u = @other_users.first
+        g = create(:group)
+        g.permissions << create(:permission, controller: 'rentals', action: 'assign_anyone')
+        g.save
+        u.groups << g
+        u.save
+        current_user(u) # set current_user to u in teh controller
```
Comment (owlee) [inline spec/controllers/rentals_controller_spec.rb]: While we're at typos 'teh'

**Comment by tomecho** [comment]
make sure to verify in create if they can select a certain user

File: spec/controllers/rentals_controller_spec.rb (hunk 3 above)
Comment (tomecho) [inline spec/controllers/rentals_controller_spec.rb]: It's artistic license

**Comment by tomecho** [comment]
added additional functionality on rental create which checks if the creator can assign the renter

File: spec/controllers/rentals_controller_spec.rb [hunk 4]
```diff
@@ -107,19 +144,23 @@
...
-        u = create(:user)
-        g = create(:group)
-        g.permissions << create(:permission, controller: 'rentals', action: 'cost_adjustment')
-        g.save
-        u.groups << g
-        u.save
+        u = create :user, groups: [
+          create(:group, permissions: [
+                   create(:permission, controller: 'rentals', action: 'cost_adjustment'),
+                   create(:permission, controller: 'rentals', action: 'assign_anyone')
```
Comment (owlee) [inline spec/controllers/rentals_controller_spec.rb]: 👍

**Pull request merged**